b2/src/test/java/org/jclouds/b2/features/ObjectApiLiveTest.java

```diff
@@ -0,0 +1,301 @@
+         assertThat(payload.getContentMetadata().getContentType()).isEqualTo(contentType);
+
+         InputStream actual = null;
+         InputStream expected = null;
+         try {
+            actual = payload.openStream();
+            expected = byteSource.openStream();
+            assertThat(actual).hasContentEqualTo(expected);
+         } finally {
+            Closeables2.closeQuietly(expected);
+            Closeables2.closeQuietly(actual);
+         }
+      } finally {
+         if (uploadFile != null) {
+            objectApi.deleteFileVersion(uploadFile.fileName(), uploadFile.fileId());
+         }
+         bucketApi.deleteBucket(response.bucketId());
+      }
+   }
+
+   @Test(groups = "live")
+   public void testDownloadFileByName() throws Exception {
+      BucketApi bucketApi = api.getBucketApi();
+      ObjectApi objectApi = api.getObjectApi();
+
+      ByteSource byteSource = TestUtils.randomByteSource().slice(0, 1024);
+      Payload payload = Payloads.newByteSourcePayload(byteSource);
+      payload.getContentMetadata().setContentLength(byteSource.size());
+      String fileName = "file name";  // intentionally using spaces in file name
+      String contentSha1 = byteSource.hash(Hashing.sha1()).toString();
+      String contentType = "text/plain";
+      payload.getContentMetadata().setContentType(contentType);
+      Map<String, String> fileInfo = ImmutableMap.of("author", "unknown");
+      B2Object b2Object = B2Object.create(fileName, contentSha1, fileInfo, payload);
+
+      Bucket response = bucketApi.createBucket(BUCKET_NAME, BucketType.ALL_PRIVATE);
+      UploadFileResponse uploadFile = null;
+      try {
+         UploadUrlResponse uploadUrl = objectApi.getUploadUrl(response.bucketId());
+
+         uploadFile = objectApi.uploadFile(uploadUrl, b2Object);
+
+         b2Object = objectApi.downloadFileByName(BUCKET_NAME, fileName);
+         payload = b2Object.payload();
+         assertThat(b2Object.fileName()).isEqualTo(fileName);
+         assertThat(b2Object.contentSha1()).isEqualTo(contentSha1);
+         assertThat(b2Object.fileInfo()).isEqualTo(fileInfo);
+         assertThat(b2Object.uploadTimestamp()).isAfterYear(2015);
+         assertThat(payload.getContentMetadata().getContentType()).isEqualTo(contentType);
+
+         InputStream actual = null;
+         InputStream expected = null;
+         try {
+            actual = payload.openStream();
+            expected = byteSource.openStream();
+            assertThat(actual).hasContentEqualTo(expected);
+         } finally {
+            Closeables2.closeQuietly(expected);
+            Closeables2.closeQuietly(actual);
+         }
+      } finally {
+         if (uploadFile != null) {
+            objectApi.deleteFileVersion(uploadFile.fileName(), uploadFile.fileId());
+         }
+         bucketApi.deleteBucket(response.bucketId());
+      }
+   }
+
+   @Test(groups = "live")
+   public void testDownloadFileByNameNonExistent() throws Exception {
```

Comment:
In general, I don't think we need to "live" test stuff like this. We are creating unnecessary resources just to validate a jclouds behavior that does not give us any value of "how aligned with the provider api we are". The purpose of this test is to verify that we propagate and exception, and that is alreaydy covered in the mock tests.